### DISCUSSION TITLE
Production Migration Flow

### DISCUSSION THREAD

**Discussion by kingfishx23** [prisma/prisma#10088] [Q&A]
Hi, I'm reading the docs and trying to figure out how to do migrations that is safe in production, as I've experienced Prisma asking for resetting the database and I do not want this to happen.

Question 1:
I want to ask is this flow below right?
1. update the schema in dev
2. use prisma migrate dev, in dev
3. *prisma will create a new migration folder*
4. use prisma migrate deploy in testing & prod

Question 2:
As far as I understand from the docs, using migrate deploy:
- Does not issue a warning if an already applied migration is missing from migration history
- Does not detect drift (production database schema differs from migration history end state - for example, due to a hotfix
- Does not reset the database or generate artifacts (such as Prisma Client)
- Does not rely on a shadow database

If in dev there is a conflict (drift?) that requires the database to be resetted, what to do to make sure the migrate to prod is successful?

I love prisma but im kinda worried about messing stuff up in production, can anyone please enlighten me?

Thanks alot

**Comment by austincrim** [comment]
Hey @kingfishx23 👋 , we appreciate you bringing us your concerns.

> Question 1: I want to ask is this flow below right?  

Yes! You've got the basic flow outlined accurately. If I can clarify any of the steps, let me know.

> Question 2: As far as I understand from the docs, using migrate deploy...

First off, **`prisma migrate deploy` will never reset your database**, so even if the command fails, you can be assured that your DB will not be wiped out. You can read a little more about this in our [migration FAQ](https://github.com/prisma/prisma-engines/blob/master/migration-engine/ARCHITECTURE.md#can-migrate-deploy-ask-to-reset-the-database-does-it-use-the-shadow-database).

Could you help me understand your specific concerns with the `migrate deploy` workflow? Do you feel like it doesn't provide the right guardrails?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
